mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx

```diff
@@ -171,247 +164,281 @@ interface DexItem<T> {
+      testID='your_liquidity_tab'
+      renderItem={({ item, index }: {item: {type: string, data: WalletToken}, index: number}): JSX.Element => {
+      const { data: yourPair } = item
+      const poolPairData = availablePairs.find(pr => pr.data.symbol === (yourPair as AddressToken).symbol)
+      const mappedPair = poolPairData?.data
+
+      const [symbolA, symbolB] = (mappedPair?.tokenA != null && mappedPair?.tokenB != null)
+        ? [mappedPair.tokenA.displaySymbol, mappedPair.tokenB.displaySymbol]
+        : yourPair.symbol.split('-')
+      const toRemove = new BigNumber(1).times(yourPair.amount).decimalPlaces(8, BigNumber.ROUND_DOWN)
+      const ratioToTotal = toRemove.div(mappedPair?.totalLiquidity?.token ?? 1)
+      const symbol = `${symbolA}-${symbolB}`
+
+      // assume defid will trim the dust values too
+      const tokenATotal = ratioToTotal.times(mappedPair?.tokenA.reserve ?? 0).decimalPlaces(8, BigNumber.ROUND_DOWN)
+      const tokenBTotal = ratioToTotal.times(mappedPair?.tokenB.reserve ?? 0).decimalPlaces(8, BigNumber.ROUND_DOWN)
+
+      return (
+        <ThemedView
+          dark={tailwind('bg-gray-800 border-gray-700')}
+          light={tailwind('bg-white border-gray-200')}
+          style={tailwind('mx-4 p-4 my-1 border rounded', { 'mt-4': index === 0 })}
+          testID='pool_pair_row_your'
+        >
+          <View style={tailwind('flex-row items-center justify-between')}>
+            <View style={tailwind('flex-row items-center')}>
+              <PoolPairIcon symbolA={symbolA} symbolB={symbolB} />
+              <ThemedText
+                style={tailwind('text-base font-medium')}
+                testID={`your_symbol_${symbol}`}
+              >
+                {symbol}
+              </ThemedText>
+            </View>
+          </View>
+          <PoolPairInfoDetails
+            type='your'
+            pair={mappedPair}
+            pairAmount={yourPair.amount}
+            tokenATotal={tokenATotal}
+            tokenBTotal={tokenBTotal}
+            testID='your'
+          />
+          <PoolPairActions
+            onAdd={() => onAdd((poolPairData as DexItem<PoolPairData>)?.data)}
+            onRemove={() => onRemove((poolPairData as DexItem<PoolPairData>)?.data)}
+            symbol={symbol}
+          />
+        </ThemedView>
+      )
+    }}
+    />
+)
 }
 
-function PoolPairRowAvailable ({
-  data,
-  onAdd,
-  onSwap
-}: { data: PoolPairData, onAdd: () => void, onSwap: () => void }): JSX.Element {
-  const [symbolA, symbolB] = (data?.tokenA != null && data?.tokenB != null)
-    ? [data.tokenA.displaySymbol, data.tokenB.displaySymbol]
-    : data.symbol.split('-')
-  const symbol = `${symbolA}-${symbolB}`
-
+function AvailablePoolPairCards ({
+  availablePairs,
+  onAdd
+}: { availablePairs: Array<DexItem<PoolPairData>>, onAdd: (data: PoolPairData) => void }): JSX.Element {
+  const navigation = useNavigation<NavigationProp<DexParamList>>()
```

Comment:
It's because of the `param: pair` Since `AvailablePoolPairCards` loops all the cards with a `FlatList`, imo it would be better to colocate it with where the `pair` is.